framework-cloud/framework-cloud-api/src/main/java/org/kie/cloud/api/scenario/KieDeploymentScenario.java

```diff
@@ -62,4 +64,13 @@
      * @see ControllerDeployment
      */
     List<ControllerDeployment> getControllerDeployments();
+
+    /**
+     * Return GIT provider.
+     *
+     * @return GIT provider.
+     */
+    default Optional<GitProvider> getGitProvider() {
```

Comment:
What is the motivation for using Optional here? Is empty git provider a valid scenario?

Reply:
I thought GitProvider is not always required, that's why the optional. If it's always required, we could remove the Optional and return the new GitProviderService().buildGitProvider() by default?

Reply:
> If it's always required, we could remove the Optional and return the new GitProviderService().buildGitProvider() by default?

Sounds good to me.
There has been a discussion about using Optional for such cases some time ago (when Maven was refactored). The result is that Optional is not meaningful in this case. 

The reasons behind were that this framework is not intended to be used in other framework or software, but rather by tests themselves. If the test calls getGitProvider then it expect to get GitProvider instance. Returning empty value is not a valid option for test. Therefore if the provider is not available for some reason then it is better to throw an exception (fail fast).

Reply:
It makes sense to me. If caller calls to getGitProvider() it should fail if the scenario does not support it.
Will change it soon. 
Thanks,

Reply:
Done.